### PR TITLE
🏗🚮 Clean up `package.json` by removing unused / adding missing deps

### PR DESCRIPTION
**Pull request by rsimha**
I ran [`npm-check`](https://www.npmjs.com/package/npm-check) and [`depcheck`](https://www.npmjs.com/package/depcheck) to generate a list of unused and missing dependencies in `package.json`. This PR adds / removes them as appropriate.

Unfortunately, it's not easy to add an always-on check that runs `npm-check` and `depcheck` for all PRs because the tools report several false positives for dependencies that are in fact used in build / test tools (where there is no `require` or `import` statement that mentions the dependency).

### CODE REVIEW

File: build-system/tasks/runtime-test/index.js [hunk 1]
```diff
@@ -513,7 +513,7 @@ async function runTests() {
       if (!argv.saucelabs && !argv.saucelabs_lite) {
         log(green('Running tests locally...'));
       }
-    }).on('browsers_ready', function() {
+    }).once('browsers_ready', function() {
```
Comment (danielrozenberg) [inline build-system/tasks/runtime-test/index.js]: This doesn't seem related to the PR description?

Comment (rsimha) [reply build-system/tasks/runtime-test/index.js]: It is mentioned in the description.

Comment (danielrozenberg) [reply build-system/tasks/runtime-test/index.js]: So it is! Missed it even though I read the description twice >__>

Comment (rsimha) [reply build-system/tasks/runtime-test/index.js]: Removed code and edited description.

(For context, that fix was for a rare bug where this event would sometimes trigger twice, and cause the `Done. Running tests...` message to be printed more than once.)
